[Create Test Modules] Creating BINFO-2.1 and BRBINFO-2.1 python3 test modules

j-ororke opened the pull request:
#### Summary

- Removes last two steps. See test plan for explanation.
- Created BINFO-2.1 and BRBINFO-2.1 python3 test module code is the same between both test modules, just more focused in the BRBINFO test which is why we created a support module following similar method done by Cecille with BINFO-3.1 and BRBINFO-3.1 test modules with the product_appearance support module in [41682](https://github.com/project-chip/connectedhomeip/pull/41682)
- Removes BINFO-2.1 and BRBINFO-2.1 yaml scripts
- Adds pycountry in requirements.txt for country code Location attribute validation in test step 7 for BINFO test as BRBINFO does not support that attribute.

#### Related issues

- Completes task [627](https://github.com/project-chip/matter-test-scripts/issues/627)
- Follows Test Plan PR [5710](https://github.com/CHIP-Specifications/chip-test-plans/pull/5710)

#### Testing

Testing completed in WSL with the following commands:
BINFO-2.1 command:
```
./scripts/tests/run_python_test.py --factory-reset --app out/linux-x64-all-clusters/chip-all-clusters-app --app-args "--discriminator 1234" --script-args "--storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --PICS src/app/tests/suites/certification/ci-pics-values --endpoint 0" --script src/python_testing/TC_BINFO_2_1.py 
```

BRBINFO-2.1 command:
```
 ./scripts/tests/run_python_test.py --factory-reset --app out/linux-x64-bridge/chip-bridge-app --app-args "--discriminator 1234" --script-args "--storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --PICS src/app/tests/suites/certification/ci-pics-values --endpoint 3" --script src/python_testing/TC_BRBINFO_2_1.py 
```

Note:

j-ororke committed:
Creating BINFO-2.1 and BRBINFO-2.1 python3 test modules:
- Created following similar method done by Cecille with BINFO-3.1 and BRBINFO-3.1 test modules with the product_appearance support module in [41682](https://github.com/project-chip/connectedhomeip/pull/41682)
- Completes task [627](https://github.com/project-chip/matter-test-scripts/issues/627)
- Follows Test Plan PR [5710](https://github.com/CHIP-Specifications/chip-test-plans/pull/5710)

j-ororke committed:
Updating binfo_attributes_verification support module

j-ororke committed:
Updating endpoint in CI arguments for BRBINFO-2.1 test module

j-ororke committed:
Ruff fixes

restyled-commits committed:
Restyled by autopep8

restyled-commits committed:
Restyled by isort

j-ororke committed:
Resolving linting error

cecille commented on src/python_testing/support_modules/binfo_attributes_verification.py:
```diff
@@ -0,0 +1,285 @@
+        self.step(0)  # commissioning already done
+
+        # Step 1: DataModelRevision
+        if cluster_name == "BRBINFO":
```
you can get rid of this - the attribute guard will handle this. The attribute just won't be present on bridged device basic info

cecille commented on src/python_testing/support_modules/binfo_attributes_verification.py:
```diff
@@ -0,0 +1,285 @@
+            try:
+                parsed_date = datetime.strptime(date_str, "%Y%m%d")
+                # Verify it's not a future date
+                asserts.assert_less_equal(parsed_date, datetime.now(), "ManufacturingDate should not be in the future")
```
Suggest also adding a check that this isn't before the first matter release date.

j-ororke replied:
Resolved, as discussed during tuesday morning test and infra team call, since January 1st 2020 is a valid date as long as testing against the SDK, then will allow January 1st 2020 to be a valid manufacturing date to pass this test.

cecille reviewed (commented):
minor stuff, but this should go in with these minor fixes.

j-ororke committed:
Merge branch 'master' into create_TC-BINFO-2-1_python3-test

j-ororke committed:
Updating to make code cleaner from suggestions by Cecille:
- Updated to remove cluster_name var, replaced with hasattr() to check if attributes were available in cluster in combination with attribute_guard()
- Updated to using validators package, included it in requirements.txt
- Updated to check if manufacturing date was before release date of matter spec version 1.0, added a TODO in the code as it does appear the default value needs to be updated to a later date.

restyled-commits committed:
Restyled by autopep8

restyled-commits committed:
Restyled by isort

j-ororke committed:
removed logging import as not needed during test currently

restyled-commits committed:
Restyled by ruff

j-ororke committed:
Merge branch 'master' into create_TC-BINFO-2-1_python3-test

j-ororke committed:
Updating manufacturing date validation in test step 12 to include SDK default date of January 1st 2020 as a valid date

restyled-commits committed:
Restyled by ruff

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into create_TC-BINFO-2-1_python3-test

j-ororke committed:
Updating imports for async_test_body and default_matter_test_main as recent changes were made to test framework

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Adding if_ci guard to manufacturing date check in test step 12 to make sure that the sdk_default_date is only allowed in CI and not during cert

restyled-commits committed:
Restyled by ruff

j-ororke committed:
Merge remote-tracking branch 'upstream/master' into create_TC-BINFO-2-1_python3-test
